CHANGELOG.md

```diff
@@ -8,6 +8,10 @@ Changes by Version
 
 #### Breaking Changes
 
+* Remove deprecated TLS flags ([#2790](https://github.com/jaegertracing/jaeger/issues/2790), [@albertteoh](https://github.com/albertteoh)):
+    * `--collector.grpc.tls` is replaced by `--collector.grpc.tls.enabled`
```

Comment:
I don't think this applies only to collector, many components share the tlscfg. Ideally we should list all flags that are affected.

Reply:
```
$ grx github.com/jaegertracing/jaeger/pkg/config/tlscfg . | grep -v _test
./cmd/collector/app/server/grpc.go:28:	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
./cmd/collector/app/builder_flags.go:24:	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
./cmd/agent/app/reporter/grpc/flags.go:23:	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
./cmd/agent/app/reporter/grpc/builder.go:33:	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
./cmd/query/app/flags.go:34:	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
./plugin/storage/cassandra/options.go:26:	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
./plugin/storage/es/options.go:26:	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
./pkg/kafka/auth/config.go:25:	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
./pkg/kafka/auth/options.go:21:	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
./pkg/kafka/auth/tls.go:23:	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
./pkg/cassandra/config/config.go:27:	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
./pkg/es/config/config.go:36:	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
```

Reply:
I think this is the full list (from docs repo):
```
$ grep -rn tls  data/cli/1.21/* | grep deprecated | grep usage | awk '{print $5}' | sort -u
--cassandra-archive.tls.enabled
--cassandra.tls.enabled
--collector.grpc.tls.client-ca
--collector.grpc.tls.enabled
--es-archive.tls.enabled
--es.tls.enabled
--kafka.consumer.tls.enabled
--kafka.producer.tls.enabled
--reporter.grpc.tls.enabled
```

Reply:
note that `--reporter.grpc.tls.enabled` is from the agent, others are shared by many components

Reply:
Yup, good call, I'll fix that up.

Reply:
> I think this is the full list (from docs repo)

Thanks @yurishkuro! I wouldn't have been able to find those references. I've updated the CHANGELOG based on your list.